### PR TITLE
fix angle

### PR DESCRIPTION
**Pull request by fvlntn**
seems to be working

### CODE REVIEW

**Comment by Vinjul1704** [comment]
Thanks, looks like it's working. Tested on current patch with crabs and the frog people (or something) in liurnia, as well as soldiers in limgrave.

Might want to change how it's displayed, so instead of "12.34myangle 56.78enemynangle" it could be "Player angle: 12.34° | Enemy angle: 56.78°", but I would say that's up to John.

**Comment by fvlntn** [comment]
yeah, I didn't know about displaying, so just put that up for debug

**Comment by veeenu** [comment]
Thank you so much for your contribution! 🚀🚀🚀

I will be able to do a proper review in a few days, but the calculations make sense. I was trying to avoid conditionals but most likely they are totally fine.

I think a compass like visualization would work better, but we can do that in a separate PR.

**Comment by fvlntn** [comment]
I think you could avoid conditionals by using atan2 like you did and correcting the angle to be 0 at East and going anti-clockwise like in real maths. One more thing is correcting the division by position.z - z which could be zero so one more conditional to write.

File: practice-tool/src/widgets/target.rs [hunk 1]
```diff
@@ -289,6 +317,40 @@ impl Widget for Target {
...
+            if relative_angle > 360.0 {
+                relative_angle = relative_angle - 360.0;
+            }
+
+            relative_angle = relative_angle - 180.0;
+
+            ui.text(format!("{distance:>6.3}m {relative_angle:>6.4}deg",));
+            ui.text(
+                format!("{my_angle_display:>6.4}myangle {enemy_angle_display:>6.4}enemyangle",),
+            );
```
Comment (veeenu) [inline practice-tool/src/widgets/target.rs:352]:
```suggestion
            ui.text(format!("{distance:>6.3}m {my_angle_display:>6.3}deg",));
```

Let's keep it like this for now, meanwhile I'm going to cook an angle visualization before merging this into a new release.

**Review by veeenu — CHANGES_REQUESTED** [review]
Thank you so much for this! Just a minor change request so we have a clean `main`. I'll think of an appropriate viz later.

**Review by veeenu — APPROVED** [review]

**Pull request merged**